src/data/polynomial/bernstein.lean

```diff
@@ -0,0 +1,279 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import data.polynomial.derivative
+import data.polynomial.algebra_map
+import linear_algebra.basis
+import data.nat.pochhammer
+import tactic.omega
+
+/-!
+# Bernstein polynomials
+
+The definition of the Bernstein polynomials
+`bernstein_polynomial (n ν : ℕ) : polynomial ℤ := (choose n ν) * X^ν * (1 - X)^(n - ν)`
+and the fact that for `ν : fin (n+1)` these are linearly independent over `ℚ`.
+
+## Future work
+
+The basic identities
+* `(finset.range (n + 1)).sum (λ ν, bernstein_polynomial n ν) = 1`
+* `(finset.range (n + 1)).sum (λ ν, ν • bernstein_polynomial n ν) = n • X`
+* `(finset.range (n + 1)).sum (λ ν, (ν * (ν-1)) • bernstein_polynomial n ν) = (n * (n-1)) • X^2`
+and the fact that the Bernstein approximations
+of a continuous function `f` on `[0,1]` converge uniformly.
+This will give a constructive proof of Weierstrass' theorem that
+polynomials are dense in `C([0,1], ℝ)`.
+-/
+
+/-!
+Two lemmas about polynomials that do not seem widely useful
+outside their application here to Bernstein polynomials.
+-/
+namespace polynomial
+variables {R : Type*} [comm_ring R]
+
+lemma derivative_comp_one_sub_X (p : polynomial R) :
```

Comment:
I nevertheless think they can move to the other lemmas about derivative. They don't seem that unuseful to me :wink:

Reply:
Moved.